src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -12,741 +24,687 @@ CreateSimilarityIdxCtrl.$inject = ['$scope', 'toastr', '$uibModal', '$timeout',
-                        $scope.newIndex.stopList,
-                        $scope.currentQuery.inference,
-                        $scope.currentQuery.sameAs,
-                        indexType,
-                        $scope.newIndex.analyzer)
-                        .error(function (err) {
-                            toastr.error(getError(err), $translate.instant('similarity.create.index.error'));
-                        });
-                    $location.url('similarity');
-                }
+    $scope.queryChanged = () => {
+        $scope.setDirty();
+        $scope.similarityIndexInfo.markInvalidQuery(undefined, false);
+        $scope.similarityIndexInfo.markInvalidQueryType(undefined, false);
+    }
 
-            })
-            .error((error) => {
-                const msg = getError(error);
-                toastr.error(msg, $translate.instant('similarity.could.not.get.indexes.error'));
-            });
-    };
+    $scope.isCloneViewMode = () => {
+        return SimilarityViewMode.CLONE === $scope.viewMode;
+    }
 
-    $scope.setViewType = (viewType) => {
-        $scope.viewType = viewType;
+    $scope.isEditViewMode = () => {
+        return SimilarityViewMode.EDIT === $scope.viewMode;
     }
 
-    // ============================================
-    // Yasqe Ysr setup public functions and setups
-    // ============================================
+    $scope.isCreateViewMode = () => {
+        return SimilarityViewMode.CREATE === $scope.viewMode;
+    }
 
-    // Called when user clicks on a sample query
-    $scope.setQuery = (query) => {
-        // Hack for YASQE bug
-        window.editor.setValue(query ? query : ' ');
+    $scope.setDirty = () => {
+        $scope.isDirty = true;
     };
 
-    // TODO don't copy paste each time, this is the same as in the graph config
-    // DOWN HERE WE KEEP EVERYTHING PURELY QUERY EDITOR (MOSTLY BORROWED FROM query-editor.controller.js)
-
-    $scope.showEditor = () => {
-        if (window.editor.xhr) {
-            window.editor.xhr.abort();
-        }
-        $scope.viewMode = 'yasr';
-    };
+    $scope.isYasqeShown = () => {
+        return $scope.yasguiConfig && RenderingMode.YASQE === $scope.yasguiConfig.render;
+    }
 
-    $scope.showPreview = () => {
-        // For some reason YASR gets confused and sets this to rawResponse
-        // if we execute a CONSTRUCT and then a SELECT. This makes sure it's always table.
-        $scope.currentQuery.outputType = 'table';
-        $scope.runQuery();
-    };
+    $scope.isYasrShown = () => {
+        return $scope.yasguiConfig && RenderingMode.YASR === $scope.yasguiConfig.render;
+    }
 
-    $scope.resetCurrentTabConfig = () => {
-        $scope.currentTabConfig = {
-            pageSize: 100, // page limit 100 as this is only used for preview
-            page: 1,
-            allResultsCount: 0,
-            resultsCount: 0
-        };
+    $scope.showEditor = () => {
```

Comment:
In the old implementation they call `abort()` for the running query on the editor if any. You are not doing it here. Also I think we don't expose this method from the component, but probably we should.

Reply:
When we change the configuration, the instance is destroyed. When YASGUI is destroyed, any ongoing request is aborted automatically (if any). But in the next request, I will change this function to not destroy the instance, so I will have to abort the query manually, as I mentioned.

Reply:
Just to clarify is this the `abort the request` or `abort the query`?

I would guess if the instance is destroyed it would abort the requests, but this would not abort the GDB query.

Reply:
Aborts the request.